OAuth/RequestDataStorage/SessionStorage.php

```diff
@@ -65,7 +67,11 @@ public function save(ResourceOwnerInterface $resourceOwner, $value, $type = 'tok
 
             $key = $this->generateKey($resourceOwner, $value['oauth_token'], 'token');
         } else {
-            $key = $this->generateKey($resourceOwner, \is_array($value) ? reset($value) : $value, $type);
+            $key = $this->generateKey($resourceOwner, $this->getStorageKey($value), $type);
+        }
+
+        if (\is_object($value)) {
```

Comment:
How about moving this logic to a private method, `getStoreageValue`, similar to `getStorageKey`?

Reply:
Done, thank you.